doc/developer-notes.md

```diff
@@ -839,6 +840,37 @@ namespace {
 #endif // BITCOIN_FOO_BAR_H
 ```
 
+Construct On First Use idiom
+------------------------------
+
+It is not possible to control the order in which non-local static objects in *different* translation units are initialized.
+This is the so-called "`static` initialization order problem". It could cause the use of an object:
```

Comment:
I think the markdown is more readable without wrapping "static" in code markup (same for line 853 below)